Move speech document

sairamya93 opened the pull request:
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

## What

To handle the rendering of Speech document type from government_frontend to frontend as part of [RFC 175]

## Why

https://trello.com/c/IKDY5W9u/371-move-document-type-speech-from-government-frontend-to-frontend, [Jira issue PNP-7351](https://gov-uk.atlassian.net/browse/PNP-7351)

## How

To add a scoped route for, a controller, a model and a view. Here we split up the methods in the original file into presenter, model and its related concerns, with the minimum functionality required to get Speech document working.

## Screenshots?

### Speaker without profile

Speech page in Government-frontend and frontend

![Screenshot 2025-03-04 at 16 39 13](https://github.com/user-attachments/assets/09794791-e882-4ae0-a034-207f48a2d7e6)
 
### Speaker with profile
### Frontend

![Screenshot 2025-01-09 at 10 39 43](https://github.com/user-attachments/assets/2f28032d-a484-4236-a890-8fa8e7294bbe)
![Screenshot 2025-01-09 at 10 40 22](https://github.com/user-attachments/assets/d6084723-2c86-4479-b75b-d88300198872)
![Screenshot 2025-01-09 at 10 40 53](https://github.com/user-attachments/assets/1fa0c850-10c4-4b1a-b30c-583b2bc21251)

### Government-Frontend

![Screenshot 2025-01-09 at 10 44 31](https://github.com/user-attachments/assets/4358e569-a7f5-4e2e-a57d-9a94f75d7792)
![Screenshot 2025-01-09 at 10 44 56](https://github.com/user-attachments/assets/5c07efc1-431c-42fa-a83f-fdd086220f10)
![Screenshot 2025-01-09 at 10 45 59](https://github.com/user-attachments/assets/686cbe43-6192-4405-ae0d-63127081de25)

### Metadata

![Screenshot 2025-01-22 at 15 02 21](https://github.com/user-attachments/assets/0dd0ae3d-befe-4688-91bc-ad555705035c)

sairamya93 committed:
Update README.md and govuk_examples.yml

sairamya93 committed:
Add Speech route and controller

sairamya93 committed:
Add Views for Speeches

Copied over the View from government-frontend and made appropriate changes
Commit audit trail - https://github.com/alphagov/government-frontend/blob/3c8d12788047ec9ebd5c7f044f4e3f3fce0480ca/app/views/content_items/speech.html.erb

- Updated the context and context_locale to reflect the correct translations
- Modified code to render publisher_metadata
- Modified withdrawn notice code in view
- Modified code to render important-metadata
- Audit trail for important-metadata change: https://github.com/alphagov/government-frontend/commit/b1da1974729332a7abcea81a34c39400c574ebfb
- Updated the component with /heading appropriate parameters instead of title component
- Removed the check for important_metadata
- Modified code to have important-metadata directly in View instead of in the presenter

sairamya93 committed:
Add locales for shared.historically_political

- To handle the translations in history_notice partial
- Audit trail: https://github.com/alphagov/government-frontend/tree/90cf5386b165b3a342689744feece7c83449b153/config/locales

KludgeKML commented on app/controllers/speech_controller.rb:
The indenting on this file looks a bit off (4 rather than 2 spaces) - rubocop is probably complaining about this?

deborahchua commented on app/models/speech.rb:
I think this commit should come later once you've committed the concerns included here.

deborahchua commented on app/controllers/speech_controller.rb:
```diff
@@ -0,0 +1,5 @@
+class SpeechController < ContentItemsController
+    def show
+        @presenter =  SpeechPresenter.new(@content_item)
```
The presenter and view hasn't been included yet, so if you were to load a speech page whilst on this commit it would error.
You could update this commit so that the `show` method is empty, and also include an empty view (app/views/speech/show.html.erb).

deborahchua commented on app/models/speech.rb:
```diff
@@ -0,0 +1,8 @@
+class Speech < ContentItem
+  include Body
```
I think this commit should come later once you've committed the concerns included here.

deborahchua commented on app/models/concerns/news_image.rb:
```diff
@@ -0,0 +1,23 @@
+module NewsImage
```
I think the concerns should be put in to separate commits and the corresponding tests included too e.g. for this file you can include https://github.com/alphagov/frontend/pull/4573/commits/70b58e793e21872674dac097e78959138b962816 in the same commit.

deborahchua commented on app/models/concerns/political.rb:
```diff
@@ -0,0 +1,32 @@
+module Political
```
Similarly here, you can put this concern in a separate commit and combine it with https://github.com/alphagov/frontend/pull/4573/commits/5996012547f39f30f6bf34674e18656574efebb7

deborahchua commented on app/views/shared/_history_notice.html.erb:
```diff
@@ -0,0 +1,8 @@
+<% if content_item.historically_political? %>
```
I think this shared partial should also be in a separate commit, which is then followed by the generation of the locale keys in https://github.com/alphagov/frontend/pull/4573/commits/c08cd5b92c591f63772f35078862ba2687e23b57

deborahchua commented on app/presenters/speech_presenter.rb:
```diff
@@ -0,0 +1,41 @@
+  def delivered_on_metadata
+    "#{delivered_on}#{speech_type_explanation}"
+  end
+
```
Could you update the commit message if the [from](https://github.com/alphagov/government-frontend/blob/122ae292fa540059ee165a94f8129d873c5205d5/app/presenters/speech_presenter.rb#L22-L26) and [speaker_without_profile](https://github.com/alphagov/government-frontend/blob/122ae292fa540059ee165a94f8129d873c5205d5/app/presenters/speech_presenter.rb#L50-L52) methods are no longer needed or will live somewhere else, otherwise at first glance it looks like the methods were accidentally missed.

deborahchua commented on app/presenters/speech_presenter.rb:
```diff
@@ -0,0 +1,41 @@
+    @view_context = ApplicationController.new.view_context
+  end
+
+  def display_date(timestamp, format = "%-d %B %Y")
```
You can remove this method as the helper method is already available in [date_helper.rb](https://github.com/alphagov/frontend/blob/main/app/helpers/date_helper.rb)

deborahchua commented on app/views/speech/show.html.erb:
```diff
@@ -0,0 +1,58 @@
+<% content_for :extra_headers do %>
+   <%= render "govuk_publishing_components/components/machine_readable_metadata", { content_item: @content_item.to_h, schema: :article } %>
+<% end %>
+
```
I think you need to include the `<main>` tag here, similar to https://github.com/alphagov/frontend/pull/4225/commits/c501ca1d9ec9c2263ea7c3f1903582bca2333f59#diff-2c4760c884914b406ca821f19956295c799447ffe9a1ae9e780322d1d0c256b4R9

deborahchua replied:
I've just seen the tag has been added in a later commit

deborahchua commented on app/views/speech/show.html.erb:
```diff
@@ -0,0 +1,58 @@
+  </div>
+</div>
+
+<%= render 'shared/publisher_metadata_with_logo' %>
```
This has been renamed to `publisher_metadata` in https://github.com/alphagov/frontend/pull/4372/commits/c4dceb3647efdbd076884ab221886c7f51bd74d0

deborahchua commented on app/views/speech/show.html.erb:
```diff
@@ -0,0 +1,58 @@
+<%= render 'shared/publisher_metadata_with_logo' %>
+<%= render 'shared/history_notice', content_item: @content_item %>
+<% if @content_item.withdrawn? %>
+  <%= render 'govuk_publishing_components/components/notice', @content_item.withdrawal_notice_component  %>
```
The way the withdrawal notice works has changed slightly, see https://github.com/alphagov/frontend/pull/4372/commits/6c5ed23d73e9103cb723e51d7e08bdfd29b64089#diff-de0d946d1ce699f8a4070b4d6a67cc44726747222c15044f2bc7e00e3a314743R31-R40

deborahchua commented on app/views/speech/show.html.erb:
```diff
@@ -0,0 +1,58 @@
+  <div class="govuk-grid-column-two-thirds">
+    <div class="content-bottom-margin">
+      <div class="responsive-bottom-margin">
+        <%= render 'components/important_metadata',
```
This is missing the condition which will only render the important metadata component if the information is available see - https://github.com/alphagov/government-frontend/blob/3c8d12788047ec9ebd5c7f044f4e3f3fce0480ca/app/views/content_items/speech.html.erb#L27
Does `inverse: true` and `margin_bottom: 0` no longer need to be included? https://github.com/alphagov/government-frontend/blob/3c8d12788047ec9ebd5c7f044f4e3f3fce0480ca/app/views/content_items/speech.html.erb#L30-L32

deborahchua commented on app/views/speech/show.html.erb:
```diff
@@ -0,0 +1,58 @@
+      </div>
+
+      <%= render 'components/published_dates', {
+          published: display_date(@content_item.first_published_at),
```
The variable used here is now `first_public_at`, similar to https://github.com/alphagov/frontend/pull/4372/commits/6c5ed23d73e9103cb723e51d7e08bdfd29b64089#diff-de0d946d1ce699f8a4070b4d6a67cc44726747222c15044f2bc7e00e3a314743R60

deborahchua commented on app/views/components/_important_metadata.html.erb:
```diff
@@ -0,0 +1,21 @@
+<% add_app_component_stylesheet("important-metadata") %>
```
This component was recently removed in https://github.com/alphagov/government-frontend/pull/3414 so you can drop this commit, and update the view

deborahchua commented on spec/components/important_metadata_spec.rb:
```diff
@@ -37,18 +37,18 @@ def component_name
     expect(rendered).to have_css(".app-c-important-metadata dt", text: "Case type:")
```
I think this file crept into this commit during a rebase

deborahchua commented on spec/system/speech_spec.rb:
```diff
@@ -0,0 +1,60 @@
+        expect(page).to have_text("Published 8 March 2016")
+      end
+
+      # Check important metadata
```
This test will need to be updated to use the metadata block component instead

deborahchua commented on spec/system/speech_spec.rb:
```diff
@@ -0,0 +1,60 @@
+      expect(page).to have_css(".gem-c-translation-nav")
+      expect(page).to have_text("بوريس جونسون: يمنح القرار الترخيص اللازم لمنظمة حظر الأسلحة الكيميائية لإزالة سلائف تلك الأسلحة من ليبيا تمهيدا لإتلافها في بلد ثالث. وبذلك نكون قد خففنا خطر وقوع هذه الأسلحة في أيدي الإرهابيين ")
+    end
+  end
```
I think an additional test can be added to test a withdrawn speech page, where we would expect the notice is included on the page.

deborahchua commented on app/models/concerns/withdrawable.rb:
```diff
@@ -9,5 +9,38 @@ def withdrawn?
       def withdrawal_notice
         content_store_hash["withdrawn_notice"]
       end
+
```
I think this accidentally crept in during a rebase.

sairamya93 committed:
Add News image concern and tests

- Commit audit trail: https://github.com/alphagov/government-frontend/blob/122ae292fa540059ee165a94f8129d873c5205d5/app/presenters/content_item/news_image.rb
- Commit audit trail for tests: https://github.com/alphagov/government-frontend/blob/90cf5386b165b3a342689744feece7c83449b153/test/presenters/content_item/news_image_test.rb

sairamya93 committed:
Add political concern and tests

- Commit audit trail: https://github.com/alphagov/government-frontend/blob/e7b9db33536f487991ba907502080e0625a066cc/app/presenters/content_item/political.rb
- Added political tests and included the possibility of having political being false even though it is not likely to be.
Reference doc: https://github.com/alphagov/whitehall/blob/13763a5b148ba06cb0e5be4139040dae311fa781/docs/history_mode.md#L11

sairamya93 committed:
Add history_notice partial

Commit audit trail - https://github.com/alphagov/government-frontend/blob/a7e99cf556dbc5d8b8fab5a317157136cadeb8a0/app/views/shared/_history_notice.html.erb

sairamya93 committed:
Add locales for written_statement

- Ran the command: consolidation:copy_translation[content_item.schema_name.written_statement,formats.written_statement]
Audit trail: https://github.com/alphagov/government-frontend/tree/ed44dc0a2dc1cc14174f99bdf6f1ed57f2bfc77a/config/locales

sairamya93 committed:
Add formats.oral_statement translations

- Ran the command: rake "consolidation:copy_translation[content_item.schema_name.oral_statement,formats.oral_statement]"
Audit trail: https://github.com/alphagov/government-frontend/tree/ed44dc0a2dc1cc14174f99bdf6f1ed57f2bfc77a/config/locales

leenagupte reviewed (commented):
Ramya, thanks for tagging me as a reviewer. Looking though the existing review comments, I think that @deborahchua  has it covered, and that if you action all of her recommendations, your PR will be in really good shape.

When it comes to constructing a PR, I was told to think of the PR as telling the story of the change. 
Some people like to add changes from "outside in" i.e. from the frontend (view) to the backend (model). 
I think as long as I can follow the steps you've taken, I don't mind whether you go "outside in" or "inside out" (backend to frontend). But I think something along the lines of "add a thing (model, method etc" then "use the thing" works well.

As Deb mentioned, when you're adding something brand new, as in this case, the tests should pass in every commit, and that should help to order the commits.

I think your code is there, so that's half the battle won. The question you should ask yourself is "will I be able to understand what I've changed 6 months down the road?"

sairamya93 committed:
Add request and system tests for speeches

- Added request tests and modified the system tests to rpsec tests in frontend
- Included test for verifying speaker without profile
- Commit audit trail: https://github.com/alphagov/government-frontend/blob/90cf5386b165b3a342689744feece7c83449b153/test/integration/speech_test.rb

deborahchua commented on config/locales/ar.yml:
I'm not sure where the `oral_statement` key is being used, is it still needed?

sairamya93 replied:
It was there in one of the examples for Speeches in Content_schemas. One of the test cases (withdrawn_speech) was failing due to the missing translation for oral_statement and hence I included that.

deborahchua replied:
Should there also be additional system tests for [each schema](https://github.com/alphagov/publishing-api/blob/main/content_schemas/dist/formats/speech/frontend/schema.json#L34-L42), to test it is displaying any variations of content? I think this key was only spotted missing because that document type happened to be used for the withdrawn speech schema example.

KludgeKML commented on app/models/speech.rb:
```diff
@@ -0,0 +1,7 @@
+class Speech < ContentItem  
+  include Organisations
```
These could probably be alphabetised.

KludgeKML commented on app/presenters/speech_presenter.rb:
```diff
@@ -0,0 +1,39 @@
+class SpeechPresenter < ContentItemPresenter
+  include DateHelper
+
+  def delivery_type
```
You could potentially refactor this with a guard clause? (not a blocking comment, just a suggestion):

```
def delivery_type
  return I18n.t("formats.speech.written_on") if content_item.document_type == "authored_article"

  I18n.t("formats.speech.delivered_on")
end
```

KludgeKML replied:
(Also this is only ever used in the important metadata method, you might make it private)

KludgeKML commented on app/presenters/speech_presenter.rb:
```diff
@@ -0,0 +1,39 @@
+
+  def important_metadata
+    { I18n.t("formats.speech.location") => location, delivery_type => delivered_on_metadata }
+  end
```
You could probably accept view_context as a parameter to this method (and pass it on to delivered_on_metadata) which would allow you to not have the view_context method. (What do you think, @leenagupte ?)

leenagupte replied:
Is `view_context` really needed? Can `tag.time` not be called in the view?

I think perhaps that the `delivered_on` method needs to be looked at again.

It's modelling `delivered_on_date` which could be moved to the model, and then calling a helper method on it, where as I the functionality in the `important_metadata` and `delivered_on_metadata` could just be moved to the view.

KludgeKML replied:
important_metadata is only called in one place (a view) - it could potentially be inlined there?

KludgeKML commented on app/models/concerns/political.rb:
```diff
@@ -0,0 +1,32 @@
+
+private
+
+  def political?
```
This construct is probably unnecessary - you're only looking for a true/false on whether the key exists, so it can be

`content_store_hash["details"].key?("political")`

sairamya93 replied:
Hmm we might have to check the value of the key "political" which could be either true/false? Like in 
https://www.gov.uk/api/content/government/speeches/orlando-frasers-speech-to-the-annual-public-meeting

I can modify as below?
content_store_hash["details"].key?("political") && content_store_hash.dig("details", "political")

KludgeKML replied:
Oh, in that case just do the dig, I think? If political is there, you'll get true/false. If it's not, you'll get nil (which evaluates to false)

KludgeKML replied:
Basically, I think my original key check was too simple, but you don't need to both check the key exists _and_ dig it. Dig will handle if the key doesn't exist.

sairamya93 replied:
Ah right. So the method will return false if key is not present or key is either nil/false. Otherwise it will return true.
I can just have this - 
content_store_hash.dig("details", "political")

KludgeKML commented on app/models/concerns/updatable.rb:
```diff
@@ -0,0 +1,40 @@
+
+private
+
+  def any_updates?
```
Again, you could reverse this with a guard clause:

```
def any_updates?
  return false unless public_updated_at && first_public_at

  Time.zone.parse(public_updated_at) != Time.zone.parse(first_public_at)
end
```

sairamya93 committed:
Modify method in updatable concern

- Modified any_updates? method with guard clause
- Added a missing test in Updatable concern

Add missing test for updatable concern

deborahchua commented on app/views/speech/show.html.erb:
```diff
@@ -1,37 +1,61 @@
-  <%= render 'shared/translations' %>
-  <div class="govuk-grid-column-two-thirds">
-    <%= render 'govuk_publishing_components/components/lead_paragraph', text: @content_item.description %>
+<main role="main" id="content" class="speech" <%= lang_attribute(@content_item.locale) %>>
```
The <main> tag has been globally included now (https://github.com/alphagov/frontend/pull/4414) so you don't need to include it here in the view.

deborahchua commented on app/assets/stylesheets/helpers/_publisher-metadata-with-logo.scss:
This file got renamed and was already merged in https://github.com/alphagov/frontend/pull/4372/commits/ad511e05ad4d44aea1a7b6c2aec510647502605e so I think don't think you need this commit anymore.

KludgeKML commented on spec/support/concerns/political.rb:
```diff
@@ -0,0 +1,23 @@
+    expect(described_class.new(content_store_response).historically_political?).to be(false)
+  end
+
+  it "knows it's publishing government" do
```
typo (it's -> its)

leenagupte replied:
Is it a typo? "it knows it is publishing government" 🤔

KludgeKML commented on spec/system/speech_spec.rb:
```diff
@@ -0,0 +1,70 @@
+      content_store_has_example_item("/government/speeches/speech-translated", schema: :speech, example: "speech-translated")
+    end
+
+    it "translated speech" do
```
Missing words? (`it "displays the translated speech" do`, maybe?)

KludgeKML commented on spec/system/speech_spec.rb:
```diff
@@ -0,0 +1,70 @@
+      content_store_has_example_item("/government/speeches/withdrawn-speech", schema: :speech, example: "withdrawn-speech")
+    end
+
+    it "withdrawn speech" do
```
As above - (maybe `it "displays the withdrawn speech notice" do` ?)

KludgeKML commented on app/views/speech/show.html.erb:
```diff
@@ -1,37 +1,60 @@
+<% content_for :title do %>
```
We could maybe squash this commit into "Add views for speeches"?

KludgeKML commented on app/assets/stylesheets/application.scss:
```diff
@@ -65,3 +65,8 @@ $govuk-include-default-font-face: false;
 .bank-hols .govuk-panel {
   margin-bottom: govuk-spacing(6);
 }
+
```
Ideally we'd add this in a view stylesheet rather than in application.scss. I'm also a bit suspicious of this as it's just used to set background for a component that should probably understand it's own background:

https://github.com/alphagov/frontend/pull/4573/commits/5643a6c9750fc02bb1fab93ae10db91d399359ca#diff-97f20b0578ce81eb243573451c18ee5b961816e94b2cab6146d48aa497cc1a1fR47-R55

@andysellick could you give a bit of guidance here?

sairamya93 replied:
For now, added `inverse-background` in stylesheets/helpers and imported that in application.scss

KludgeKML commented on spec/models/speech_spec.rb:
```diff
@@ -0,0 +1,48 @@
+RSpec.describe Speech do
```
Maybe a good idea to squash this into the commit where you add the implementation of the Content Model? (this is non-blocking, though. I appreciate you might want this here after all the concerns are added)

KludgeKML commented on app/views/speech/show.html.erb:
```diff
@@ -9,11 +9,14 @@
 
   <div class="govuk-grid-row">
```
Again, you could squash this commit into: Add Views for Speeches

KludgeKML commented on app/assets/stylesheets/helpers/_inverse-background.scss:
```diff
@@ -0,0 +1,4 @@
+.inverse-background {
+    background: $govuk-brand-colour;
+    color: govuk-colour("white");
```
Reminder - I don't think you need to specify the color (just the background) here.

KludgeKML commented on app/assets/stylesheets/application.scss:
```diff
@@ -66,3 +66,8 @@ $govuk-include-default-font-face: false;
 .bank-hols .govuk-panel {
   margin-bottom: govuk-spacing(6);
 }
+
+.inverse-background {
+  background: $govuk-brand-colour;
+  color: govuk-colour("white");
+}
```
I think since this has been moved into the helper you can remove it here.

leenagupte reviewed (commented):
This is looking great! 🎉  I've added some inline comments, but they're really all quite minor.

sairamya93 commented on spec/requests/speech_spec.rb:
```diff
@@ -0,0 +1,13 @@
+  end
+
+  describe "GET index" do
+    it "returns 200" do
```
Thanks for the reference test. I have added it

sairamya93 committed:
Add locales for authored_article

Ran the command : govuk-docker-run rake "consolidation:copy_translation[content_item.schema_name.authored_article,formats.authored_article]"

Audit trail: https://github.com/alphagov/government-frontend/blob/8799336fd3c96dac7c7e53cce3cd8a522beb17da/config/locales/en.yml#L101-L103
Speech document type comprises of all the below:
- speech
- authored_article
- written_statement
- oral_statement
Reference document: https://github.com/alphagov/publishing-api/blob/648bddda06ad1afaeeebe2dc5f6f97da214d14ef/content_schemas/dist/formats/speech/frontend/schema.json#L34-L42

sairamya93 commented on spec/support/concerns/political.rb:
```diff
@@ -0,0 +1,23 @@
+RSpec.shared_examples "it has historical government information" do |document_type, example_name|
+  let(:content_store_response) { GovukSchemas::Example.find(document_type, example_name:) }
+
+  it "knows it is part of historical government if it is not the current government" do
```
That was helpful to know the context.

sairamya93 commented on app/models/concerns/political.rb:
Yes it was but I have now updated the commit message

sairamya93 commented on spec/support/concerns/news_image.rb:
```diff
@@ -0,0 +1,50 @@
+RSpec.shared_examples "it has news image" do |schema|
+  let(:content_store_response) { GovukSchemas::Example.find(schema, example_name: schema) }
+
+  before do
+    {
+      "base_path" => "/a/base/path",
+      "details" => {},
+      "links" => {},
+    }
+  end
```
Hmm yes don't think it is required.

sairamya93 commented on spec/support/concerns/news_image.rb:
```diff
@@ -0,0 +1,50 @@
+          { "details" => { "default_news_image" => default_news_image } },
+        ],
+      },
+    }
```
Ok I have updated the `before` block with the code and the individual tests will modify the values where required.

sairamya93 commented on app/models/speech.rb:
```diff
@@ -0,0 +1,31 @@
+class Speech < ContentItem
+  include NewsImage
```
Yes but I have now rebased to include the concern and model tests in their corresponding commits

leenagupte reviewed (dismissed):
This looks great Ramya!  
Thanks for making all of the requested changes.
The commit history and the audit trail are great and easy to follow too. 🥇

leenagupte commented on app/models/speech.rb:
```diff
@@ -1,11 +1,12 @@
 
   def contributors
-    (organisations_ordered_by_emphasis + [speaker].flatten + [speaker_without_profile].flatten).compact
+    (organisations_ordered_by_emphasis + people + [speaker].flatten + [speaker_without_profile].flatten).compact.uniq { |contributor| contributor["content_id"] }
```
> `{ |contributor| contributor["content_id"] }

What is this block supposed to do?

leenagupte replied:
Is `speaker` or `speaker_without_profile` an array? If they are, then they don't need to be made into an array by wrapping them in `[]`. If they aren't arrays, then `flatten` isn't required because the array will only contain one item.

leenagupte reviewed (commented):
Thanks for adding `organisations_with_emphasis` and people to the contributors. 🎉 

There a couple of inline comments.

sairamya93 committed:
Add locales for Speech to handle pluralization

- To handle the pluralization for Speech in all the locales
Run the command: rake "consolidation:copy_translation[content_item.schema_name.speech,formats.speech]"

Commit audit trail: https://github.com/alphagov/government-frontend/tree/122ae292fa540059ee165a94f8129d873c5205d5/config/locales

sairamya93 committed:
Update locales for Speech

- All the locale keys for Speech are copied over from government-frontend
- Ran the command : rake "consolidation:copy_translation[speech,formats.speech]"
Audit trail: https://github.com/alphagov/government-frontend/blob/ed44dc0a2dc1cc14174f99bdf6f1ed57f2bfc77a/config/locales

sairamya93 committed:
Add Speech model and presenter

- Added Speech model that includes the related concerns for Speech along with the tests
- Modified Speech model to include the methods to retrieve the important-metadata information that were copied over from Speech Presenter in government-frontend
- Added Speech Presenter and tests to handle contributor links based on speaker without profile and delivery_type of the speech
Commit audit trail - https://github.com/alphagov/government-frontend/blob/122ae292fa540059ee165a94f8129d873c5205d5/app/presenters/speech_presenter.rb
Audit trail for presenter tests - https://github.com/alphagov/government-frontend/blob/122ae292fa540059ee165a94f8129d873c5205d5/test/presenters/speech_presenter_test.rb

- Added People concern and tests

As there is a possibility to have people but no speaker, the people concern is included in Speech model.
Audit trail: https://github.com/alphagov/government-frontend/blob/6285583f2385567f785992fd7901209d8ddb7980/app/presenters/content_item/linkable.rb#L4

KludgeKML commented on spec/support/concerns/political.rb:
```diff
@@ -0,0 +1,28 @@
+    expect(described_class.new(content_store_response).historically_political?).to be(false)
+  end
+
+  it "knows it is publishing government" do
```
grammar:  `it "knows its publishing government" do`

KludgeKML reviewed (approved):
LGTM! (one tiny note about grammer in a test, but otherwise good)
